lib/spack/spack/build_systems/intel.py

```diff
@@ -1332,5 +1333,35 @@ def uninstall_ism(self):
         debug_print(os.getcwd())
         return
 
+    @property
+    def base_lib_dir(self):
+        """Provide the library directory located in the base of Intel installation.
+        """
+        d = self.normalize_path('')
+        d = os.path.join(d, 'lib')
+
+        debug_print(d)
+        return d
+
+    @run_after('install')
+    def modify_LLVMgold_rpath(self):
+        """Add libimf.so and other required libraries to the RUNPATH of LLVMgold.so.
+
+        These are needed explicitly at dependent link time when
+        `ld -plugin LLVMgold.so` is called by the compiler.
+        """
+        if self._has_compilers:
+            LLVMgold_libs = find_libraries('LLVMgold', self.base_lib_dir,
+                                           shared=True, recursive=True)
+            # Ignore ia32 entries as they mostly ignore throughout the rest
+            # of the file.
+            # The first entry in rpath preserves the original, the seconds entry
+            # is the location of libimf.so. If this relative location is changed
+            # in compiler releases, then we need to search for libimf.so instead
+            # of this static path.
+            rpath = "$ORIGIN/../lib:$ORIGIN/../compiler/lib/intel64_lin"
```

Comment:
One concern here is that the original rpath might not be $ORIGIN/../lib in all versions. The risk of a problem is low because oneapi replaced parallel studio > 1 year ago so there will not be any parallel studio updates that might change the path.

Reply:
OK, I can get the original rpath from `LLVMgold.so` itself. Do you think we can assume `readelf` (e.g. binutils) is present on the OS or do we need another dependency?

Reply:
You can use patchelf:
```
(venv) rscohn1@rscohn1-mobl1:spack$ patchelf --print-rpath $(spack location -i intel-parallel-studio)/compilers_and_libraries_\
2020.1.217/linux/lib/LLVMgold.so
$ORIGIN/../lib:$ORIGIN/../compiler/lib/intel64_lin
(venv) rscohn1@rscohn1-mobl1:spack$
```

Reply:
Done.